backend/src/services/post.service.js

```diff
@@ -70,3 +70,17 @@ exports.createPost = async ({
     message: 'Post is created',
   };
 };
+
+exports.getPostDetail = async ({ communityId, postId }) => {
+  const post = await Post.findById(postId).lean();
+  await Post.findOneById(postId).populate(["creator"])
+  const community = await Community.findById(communityId).lean();
+
+  if (!post) {
+    throw new ApiError(httpStatus.NOT_FOUND, 'Post does not exist');
+  }
+  if (post.communityId!=communityId){
+    throw new ApiError(httpStatus.NOT_FOUND, 'Community ID does not match');
+  }
```

Comment:
```suggestion
  const post = await Post.findById(postId).populate(["creator","community"]).lean();
  if (!post) {
    throw new ApiError(httpStatus.NOT_FOUND, 'Post does not exist');
  }
  if (post.communityId!=communityId){
    throw new ApiError(httpStatus.NOT_FOUND, 'Community ID does not match');
  }
```
We need to chain populate operations. Previous 76th line wouldn't add creator object to post, actually it would just return the document without getting assigned